src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs

```diff
@@ -161,32 +152,53 @@ impl CloudHypervisorInner {
         Ok(())
     }
 
-    pub(crate) async fn get_shared_fs_devices(&mut self) -> Result<Option<Vec<FsConfig>>> {
-        let pending_root_devices = self.pending_devices.take();
-
-        let mut root_devices = Vec::<FsConfig>::new();
+    pub(crate) async fn get_shared_devices(
+        &mut self,
+    ) -> Result<(Option<Vec<FsConfig>>, Option<Vec<NetConfig>>)> {
+        let mut shared_fs_devices = Vec::<FsConfig>::new();
+        let mut network_devices = Vec::<NetConfig>::new();
+
+        while let Some(dev) = self.pending_devices.pop() {
+            match dev {
+                DeviceType::ShareFs(dev) => {
+                    let settings = ShareFsSettings::new(dev.config, self.vm_path.clone());
```

Comment:
@amshinde, with the go runtime we're passing an FD to be used, not doing so will make SELinux integration way harder in the future.

Would you mind to follow what's done there?

Reply:
Sure, i'll follow up on this.